radixdlt/src/integration/java/com/radixdlt/consensus/BFTTest.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.consensus;
+
+import com.radixdlt.consensus.checks.AllProposalsHaveDirectParentsCheck;
+import com.radixdlt.consensus.checks.LivenessCheck;
+import com.radixdlt.consensus.checks.NoSyncExceptionCheck;
+import com.radixdlt.consensus.checks.NoTimeoutCheck;
+import com.radixdlt.consensus.checks.SafetyCheck;
+import com.radixdlt.crypto.ECKeyPair;
+import com.radixdlt.middleware2.network.TestEventCoordinatorNetwork;
+import io.reactivex.rxjava3.core.Observable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class BFTTest {
+	private final int numNodes;
+	private final long time;
```

Comment:
It seems cleaner to me to extract the configuration properties (i.e. everything except `checks`) into some BFTTestConfiguration instance and have a builder for that, which would enable us to test the same checks against various configurations more easily. Might be premature, that's for you to judge :)

Reply:
yeah will do that later methinks